packages/external-compile/index.js

```diff
@@ -260,15 +260,34 @@ const compile = callbackify(async function(options) {
   debug("running compile command: %s", command);
   await runCommand(command, { cwd, logger });
 
-  return await processTargets(targets, cwd, logger);
-});
-
-// required public interface
-compile.all = compile;
-compile.necessary = compile;
+  const contracts = await processTargets(targets, cwd, logger);
+  return [
+    {
+      contracts,
```

Comment:
Ok, so it looks like these will be just `CompiledArtifact`s parsed into a pojo. So that is not quite a `LegacyToNew`, is it? Is there a shim in `@truffle/contract` or something that deals with this when instantiating contract objects?

Reply:
Yep, but since this is external-compile, it shouldn't return an artifact, it should return `Compilation[]`, which uses the new contract format.

Reply:
Yep, totally correct.